### PR TITLE
71 targets aggregates

### PR DESCRIPTION
**Pull request by ngaruko**
# Description

https://github.com/medic/cht-release-testing/issues/71

medic/cht-core#[number]

# Code review checklist

- [ ] Readable: Concise, well named, follows the [style guide](https://docs.communityhealthtoolkit.org/contribute/code/style-guide/), documented if necessary.
- [ ] Documented: Configuration and user documentation on [cht-docs](https://github.com/medic/cht-docs/)
- [ ] Tested: Unit and/or e2e where appropriate
- [ ] Internationalised: All user facing text
- [ ] Backwards compatible: Works with existing data and configuration or includes a migration. Any breaking changes documented in the release notes.

# License

The software is provided under AGPL-3.0. Contributions to this project are accepted under the same license.

### CODE REVIEW

File: tests/page-objects/analytics/analytics.wdio.page.js [hunk 1]
```diff
@@ -0,0 +1,45 @@
+const listModule = module => $(`.mm-navigation-menu li a[href="${module}"]`);
```
Comment (newtewt) [inline tests/page-objects/analytics/analytics.wdio.page.js:1]: There seems to be a lot of unused code in this PR. Can you remove what's not used?

File: tests/e2e/targets/targets-aggregates.wdio-spec.js [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+
+  it('Supervisor Can view aggregate List', async () => {
+    await (await analyticsPage.analytics())[1].click();
+    const aggregates = await analyticsPage.getAllAggregates();
+    expect(aggregates).toEqual(['New pregnancies', 'Live births', 'Active pregnancies', 'In-facility deliveries']);
+  });
+
+  it('Supervisor Can view aggregate Details', async () => {
+    await (await analyticsPage.targetAggregatesItems())[0].click();
+    expect(await (await analyticsPage.aggregateHeading()).getText()).toBe('New pregnancies');
```
Comment (newtewt) [inline tests/e2e/targets/targets-aggregates.wdio-spec.js:81]: Using the wdio expect for `toHaveText` you can get rid of the call to `getText()`


`expect(await analyticsPage.aggregateHeading()).toHaveText('New pregnancies');`

https://webdriver.io/docs/api/expect-webdriverio/#tohavetext

**Review by newtewt — CHANGES_REQUESTED** [review]
Nice! I've added a few things in line. Sadly we cannot merge this until we can reliably get past the race condition. 

https://github.com/medic/cht-core/issues/7242

File: tests/e2e/targets/targets-aggregates.wdio-spec.js [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+const userFactory = require('../../factories/cht/users/users');
+const places = place.generateHierarchy();
+const healthCenter = places.find((place) => place.type === 'health_center');
+const clinic = places.find((place) => place.type === 'clinic');
+const analyticsPage = require('../../page-objects/analytics/analytics.wdio.page.js');
+
+healthCenter.name = 'HC_' + Date.now();
+
+const contact = personFactory.build({
+  name: 'contact_' + Date.now(),
```
[the remark this answers is not in the document]
Comment (ngaruko) [reply tests/e2e/targets/targets-aggregates.wdio-spec.js:15]: Actually thinking of extending this for the next use case and create multiple chws but for this pr, the default suffices.

**Comment by ngaruko** [comment]
Automated [here](https://github.com/medic/cht-core/blob/master/tests/e2e/target-aggregates.spec.js)

**Pull request closed without merging**